File: JavaStudy/src/main/java/SeniorStudy/CollectionTest/SetTest/Test2.java
package SeniorStudy.CollectionTest.SetTest;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;

/**
 * @ClassName Test2
 * @Description
 * @Author SDY
 * @Date 2023/8/28 21:21
 **/
public class Test2 {
    public static void main(String[] args) {
        List list = new ArrayList<>();
        list.add(new Integer(1));
        list.add(new Integer(2));
        list.add(new Integer(2));
        list.add(new Integer(4));
        list.add(new Integer(4));
        List list2 = duplicateList(list);
        for (Object o : list2) {
            System.out.println(o);
        }

    }

    public static List duplicateList(List list){
        HashSet set = new HashSet<>();
        set.addAll(list);
        return new ArrayList(set);
    }


}

File: FlinkStudy/src/main/scala/Test/Test2.scala
package Test

import UserFunction.MyFlatMapFunction
import org.apache.flink.api.common.serialization.SimpleStringSchema
import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment, createTypeInformation}
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer

import java.util.Properties

/**
 * @ClassName: Test2
 * @Auther: SDY
 * @Description:
 * @Date: 2024/12/17 10:51
 * */
object Test2 {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment

    val properties = new Properties()
    properties.setProperty("bootstrap.servers","192.168.10.101:9092,192.168.10.102:9092,192.168.10.103:9092")
    properties.setProperty("group.id","sdyTest1")
    properties.setProperty("auto.offset.reset","latest")

    val consumer: FlinkKafkaConsumer[String] = new FlinkKafkaConsumer[String]("test1", new SimpleStringSchema(), properties)
    val dataSource = env.addSource(consumer).setParallelism(2)


    val ds2: DataStream[String] = dataSource.flatMap(new MyFlatMapFunction).setParallelism(3)

    val ds3: DataStream[(String, Int)] = ds2.map(x => (x, 1)).setParallelism(4)

    val value = ds3.keyBy(_._1).sum(1).setParallelism(4)
    value.slotSharingGroup("group-1")
    value.print("result:").setParallelism(3)

    env.execute("test1")

  }

}
